tests/test_interpolation_cubic.py

```diff
@@ -0,0 +1,269 @@
+        grid = grid or Grid(shape=shape, extent=(shape[0] - 1, shape[1] - 1))
+    else:
+        grid = grid or Grid(shape=shape, extent=(shape[0] - 1,
+                            shape[1] - 1, shape[-1] - 1))
+
+    a = Function(name=name, grid=grid, space_order=0)
+    a = init_data(a)
+    return a
+
+
+def unit_box_time(name='a', shape=(11, 11),):
+    """Create a field with value 0. to 1. in each dimension"""
+    grid = Grid(shape=shape)
+    a = TimeFunction(name=name, grid=grid, time_order=1)
+    dims = tuple([np.linspace(0., 1., d) for d in shape])
+    a.data[0, :] = np.meshgrid(*dims)[1]
+    a.data[1, :] = np.meshgrid(*dims)[1]
+    return a
+
+
+def exp_box_time(name='a', shape=(11, 11), grid=None):
+    """Create a field with exponential values"""
+    def init_data(f):
+        for i in range(f.data.shape[1]):
+            for j in range(f.data.shape[2]):
+                f.data[0, i, j] = (grid.spacing[1]*j)**2
+                f.data[1, i, j] = (grid.spacing[1]*j)**2
+        return f
+
+    if len(shape) == 2:
+        grid = grid or Grid(shape=shape, extent=(shape[0] - 1, shape[1] - 1))
+    else:
+        grid = grid or Grid(shape=shape, extent=(shape[0] - 1,
+                            shape[1] - 1, shape[-1] - 1))
+
+    a = TimeFunction(name=name, grid=grid, time_order=1)
+    a = init_data(a)
+    return a
+
+
+def points(grid, ranges, npoints, name='points', cubic=None, sinc=None):
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    points = SparseFunction(name=name, grid=grid, npoint=npoints, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+def time_points(grid, ranges, npoints, name='points', nt=10, cubic=None, sinc=None):
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    points = SparseTimeFunction(name=name, grid=grid, npoint=npoints,
+                                nt=nt, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+def custom_points(grid, ranges, npoints, name='points', cubic=None, sinc=None):
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    scale = Dimension(name="scale")
+    dim = Dimension(name="dim")
+    points = SparseFunction(name=name, grid=grid, dimensions=(scale, dim),
+                            shape=(3, npoints), npoint=npoints, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+@pytest.mark.parametrize('shape, coords', [
+    ((101, 101), [(3.4, 95.), (3., 92)]),
+    ((101, 101, 101), [(3.4, 95.), (3., 92), (10., 10.)])
+])
+def test_interpolate(shape, coords, npoints=20):
```

Comment:
Can this all be gathered under a common test class?

Also I don't think it needs a new file, `test_interpolation.py` exists for this